mvcc/backend/read_tx.go

```diff
@@ -117,4 +119,84 @@ func (rt *readTx) reset() {
 	rt.buf.reset()
 	rt.buckets = make(map[string]*bolt.Bucket)
 	rt.tx = nil
+	rt.txWg = new(sync.WaitGroup)
+}
+
+// TODO: create a base type for readTx and concurrentReadTx to avoid duplicated function implementation?
```

Comment:
bundling `txMu`, `tx`, `buckets` and `txWg` into a struct might make sense. Together they form some kind of "boltdb read handler" that's got a separate lifecycle from the `buf`.

Reply:
+1

Reply:
Added TODO, will address in future PRs.